components/role-mgt/org.wso2.carbon.identity.role.mgt.core/src/main/java/org/wso2/carbon/identity/role/mgt/core/dao/RoleDAOImpl.java

```diff
@@ -1482,6 +1512,28 @@ public Set<String> getSystemRoles() {
         return systemRoles;
     }
 
+    @Override
+    public int getRolesCount(String tenantDomain) throws IdentityRoleManagementException {
+
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        try (Connection connection = IdentityDatabaseUtil.getUserDBConnection(false)) {
+            String databaseProductName = connection.getMetaData().getDatabaseProductName();
+            try (NamedPreparedStatement statement = new NamedPreparedStatement(connection,
+                    getDBTypeSpecificRolesCountQuery(databaseProductName), RoleTableColumns.UM_ID)) {
```

Comment:
Do we need to pass a returning column name for NamedPreparedStatement here?

Reply:
Fixed with 62f9bb64b2fdb14ee53e0a37cb1d9e7bf33a70d7